src/main/java/fi/oph/akt/onr/OnrServiceMock.java

```diff
@@ -69,32 +66,58 @@ public static HenkiloDto createHenkiloDto(String oid) {
 			secondName = womenSecondNames[rand.nextInt(womenSecondNames.length)];
 		}
 
-		HenkiloDto henkiloDto = HenkiloDto.builder().oidHenkilo(oid)
-				.hetu(identityNumbers[rand.nextInt(identityNumbers.length)]).etunimet(nickname + " " + secondName)
-				.kutsumanimi(nickname).sukunimi(surnames[rand.nextInt(surnames.length)])
-				.aidinkieli(new KielisyysDto("fi", "suomi")).build();
+		//@formatter:off
+		HenkiloDto henkiloDto = HenkiloDto.builder()
+				.oidHenkilo(oid)
+				.hetu(identityNumbers[rand.nextInt(identityNumbers.length)])
+				.etunimet(nickname + " " + secondName)
+				.kutsumanimi(nickname)
+				.sukunimi(surnames[rand.nextInt(surnames.length)])
+				.build();
+		//@formatter:on
 
 		henkiloDto.setSyntymaaika(getBirthDateByIdentityNumber(henkiloDto.getHetu()));
 
-		Set<YhteystietoDto> yhteystiedot = new HashSet<>();
-		yhteystiedot
-				.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_KATUOSOITE, streets[rand.nextInt(streets.length)]));
-		yhteystiedot.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_POSTINUMERO,
-				postalCodes[rand.nextInt(postalCodes.length)]));
-		yhteystiedot.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_KAUPUNKI, towns[rand.nextInt(towns.length)]));
-		yhteystiedot.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_MAA, "Suomi"));
-		yhteystiedot.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_SAHKOPOSTI,
-				henkiloDto.getKutsumanimi() + "." + henkiloDto.getSukunimi() + "@test.fi"));
-		yhteystiedot.add(new YhteystietoDto(YhteystietoType.YHTEYSTIETO_PUHELINNUMERO,
+		henkiloDto.setYhteystiedotRyhma(new HashSet<>());
+		henkiloDto.getYhteystiedotRyhma().add(createAktContactDetails(henkiloDto, rand));
+		henkiloDto.getYhteystiedotRyhma().add(createVtjContactDetails(rand));
+
+		return henkiloDto;
+	}
+
+	private static ContactDetailsGroupDto createAktContactDetails(HenkiloDto henkiloDto, Random rand) {
+		ContactDetailsGroupDto detailsGroup = new ContactDetailsGroupDto();
+		Set<ContactDetailsDto> detailsSet = new HashSet<>();
+
+		detailsSet.add(new ContactDetailsDto(YhteystietoTyyppi.YHTEYSTIETO_SAHKOPOSTI,
+				henkiloDto.getKutsumanimi().toLowerCase() + "." + henkiloDto.getSukunimi().toLowerCase() + "@akt.fi"));
+		detailsSet.add(new ContactDetailsDto(YhteystietoTyyppi.YHTEYSTIETO_PUHELINNUMERO,
 				"+35840" + (1000000 + rand.nextInt(9000000))));
 
-		YhteystiedotRyhmaDto yhteystiedotRyhma = new YhteystiedotRyhmaDto();
-		yhteystiedotRyhma.setYhteystieto(yhteystiedot);
+		detailsGroup.setType(ContactDetailsGroupType.APPLICATION_ADDRESS);
```

Comment:
Tässä näkee kuinka harhaanjohtava tuo APPLICATION_ADDRESS on, minusta tämä ei mene nyt oikein. Ainakin täytyy varmistaa että saamme näin tehdä, varminta olisi tehdä kuten OTR eli lisätä AKT:lle oma yhteystietotyyppi.